app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt

```diff
@@ -3146,6 +3170,12 @@ class BrowserTabFragment :
                     } else {
                         showHome()
                     }
+                } else {
+                    if (viewState.browserError != OMITTED) {
+                        showError(viewState.browserError)
+                    } else {
+                        showBrowser()
+                    }
```

Comment:
Maybe we could replace all if/else statements from line 3167 for when statement instead of adding more else {if/else}. I think it could improve readability